azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/service/sdk/CallingSDKWrapper.kt

```diff
@@ -122,29 +124,43 @@ internal class CallingSDKWrapper(
         callingSDKEventHandler.getRemoteParticipantInfoModelFlow()
 
     override fun hold(): CompletableFuture<Void> {
-        val call: Call?
+        val completableFuture = CompletableFuture<Void>()
 
         try {
-            call = this.call
+            val call = this.call
+            call.hold().whenComplete { _, error ->
+                if (error != null) {
+                    completableFuture.completeExceptionally(error)
+                } else {
+                    completableFuture.complete(null)
+                }
+            }
         } catch (e: Exception) {
             // We can't access the call currently, return a no-op and exit
-            return CompletableFuture.runAsync { }
+            completableFuture.completeExceptionally(e)
         }
 
-        return call.hold()
+        return completableFuture
     }
 
     override fun resume(): CompletableFuture<Void> {
-        val call: Call?
+        val completableFuture = CompletableFuture<Void>()
```

Comment:
Why do we need this refactoring with new `completableFuture`? `call.resume()` already returns one.

Reply:
To avoid a race condition crash, if Contoso calls hold resume, while composite in destroy, try catch handles it and throw exception